lib/chef/provisioning/docker_driver/docker_transport.rb

```diff
@@ -179,11 +179,12 @@ def upload_file(local_path, path)
     def make_url_available_to_remote(url)
       # The host is already open to the container.  Just find out its address and return it!
       uri = URI(url)
+      uri.scheme = 'http' if 'chefzero' == uri.scheme && uri.host == 'localhost'
       host = Socket.getaddrinfo(uri.host, uri.scheme, nil, :STREAM)[0][3]
       Chef::Log.debug("Making URL available: #{host}")
 
       if host == '127.0.0.1' || host == '::1'
```

Comment:
@tyler-ball `getaddrinfo()[0][3]` is the IP address. http://ruby-doc.org/stdlib-1.9.3/libdoc/socket/rdoc/Socket.html#method-c-getaddrinfo